### DISCUSSION TITLE
Rendering to frame buffer

### DISCUSSION THREAD

**Discussion by SasLuca** [thorvg/thorvg#2386] [Q&A] [answered]
Does anyone have an example of how to render using GlCanvas to a framebuffer?

I tried doing it but for some reason it wouldn't work so I am curious if anyone else has an example.

**Comment by hermet** [comment, marked as answer]
Hello, please have a look at this api: you can pass your `framebuffer id` to the `id` parameter.
```
    /**
     * @brief Sets the drawing target for rasterization.
     *
     * This function specifies the drawing target where the rasterization will occur. It can target
     * a specific framebuffer object (FBO) or the main surface.
     *
     * @param[in] id The GL target ID, usually indicating the FBO ID. A value of @c 0 specifies the main surface.
     * @param[in] w The width (in pixels) of the raster image.
     * @param[in] h The height (in pixels) of the raster image.
     *
     * @warning This API is experimental and not officially supported. It may be modified or removed in future versions.
     * @warning Drawing on the main surface is currently not permitted. If the identifier (@p id) is set to @c 0, the operation will be aborted.
     *
     * @see Canvas::viewport()
     *
     * @note Currently, this only allows the GL_RGBA8 color space format.
     * @note Experimental API
    */
    Result GlCanvas::target(int32_t id, uint32_t w, uint32_t h) noexcept;
```

There are plenty of examples in `thorvg/examples`. You can find out how they create a GL canvas with framebuffer usage.
![image](https://github.com/thorvg/thorvg/assets/3711518/8a6b7988-6bbc-413f-9a0c-6d0c6fe99c4a)

\+ @RuiwenTang

**Reply by SasLuca** [reply]
Thanks for the reply.

Indeed that api works as intended. My issue is, I am using 2 opengl contexts with a separate one for thorvg, and I couldn't access the framebuffer texture from the thorvg context in my other context.

In my case this was solved by calling `SDL_GL_SetAttribute(SDL_GL_SHARE_WITH_CURRENT_CONTEXT, 1);` which enables sharing resources between opengl contexts in SDL. Hope this question may be of use to others in the future as well.
